services/nsn/modules/nodejs/nsn-get-routing/src/processor/nsn-routing-create-processor.ts

```diff
@@ -13,14 +13,29 @@ export const postNsn = async (event: APIGatewayProxyEvent, context: Context): Pr
     }
     console.log('1 body - ' + event.body);
 
-    const { routing_id, owa, is_civ_mgr, is_mil_mgr, ric, created_by } = JSON.parse(event.body);
-
+    let { routing_id, owa, is_civ_mgr, is_mil_mgr, ric, created_by } = JSON.parse(event.body);
+    
     if (!routing_id) {
-        return apiResponses._400({ message: 'Routing is mandetory to create NSN record' });
+        return apiResponses._400({ message: 'Routing is mandatory to create NSN record.' });
     }
+    routing_id = routing_id.trim(); // trimming routing id for extra spaces
     if (isNaN(routing_id.substring(0, 4))) {
-        return apiResponses._400({ message: 'Routing id should be numeric' });
+        return apiResponses._400({ message: 'Please check routing Id restrictions.' });
+    }
+    if (routing_id.length<4 && routing_id.length!=2) {
+        return apiResponses._400({ message: 'Invalid routing Id, please check routing Id restrictions.' });
+    }
+    if (routing_id.length>15) {
+        return apiResponses._400({ message: 'Routing id can not be more than 15 characters.' });
     }
+    // Setting valid values for Civ and Mil Manager
+    is_civ_mgr = (is_civ_mgr=="Y")?is_civ_mgr:"N";
+    is_mil_mgr = (is_mil_mgr=="Y")?is_mil_mgr:"N";
```

Comment:
Can we do something like this?
is_civ_mgr = !is_civ_mgr ? 'N' : is_civ_mgr;
Also, I am using single quotes to denote string. Please let me know if we need to use double quotes instead. I think we should just follow same standard all over.